### PR TITLE
Polo: cancelOrder: mark order as 'canceled' when cancellation request pending

### PR DESCRIPTION
**Pull request by mkutny**
I think it's not right that 'pending' exception is a subclass of `OrderNotFound`.

How we deal with it on a client side:
- cancelOrder (id) -> receive NetworkError
- in the meantime, exchange received the request and successfully executed it
- we retry -> cancelOrder (id) -> receive OrderNotFound this time

Once we've got `OrderNotFound` we're confident that this order is not opened (either was canceled by us or executed).

But pending states are not that predictable. Right now our expectation is that `pending` state will resolve with success eventually and usually it does. But what if `pending` state does not succeed?

As to me, these pending exceptions should be a subclass of a new `TransientState` class which indicates that we need to retry later, much like we handle `NetworkErrors`.

### CODE REVIEW

File: js/poloniex.js [hunk 1]
```diff
@@ -683,15 +683,25 @@ module.exports = class poloniex extends Exchange {
             response = await this.privatePostCancelOrder (this.extend ({
                 'orderNumber': id,
             }, params));
-            if (id in this.orders)
-                this.orders[id]['status'] = 'canceled';
         } catch (e) {
-            if (this.last_http_response) {
-                if (this.last_http_response.indexOf ('Invalid order') >= 0)
-                    throw new OrderNotFound (this.id + ' cancelOrder() error: ' + this.last_http_response);
+            if (e instanceof CancelPending) {
```
Comment (kroitor) [inline js/poloniex.js]: This instanceof will cause some troubles with the the transpiler... The flow of RTTI for exceptions in JS is different from Python/PHP/C++, where type matching is done within the catch clause. This is why we can't use instanceof like that, unfortunately... Unless we add a case for it to the transpiler. Let me investigate, plz. Will get back to you on this shortly.

Comment (mkutny) [reply js/poloniex.js]: Oops... Yeah, you told me that already, I completely forgot (

Comment (kroitor) [reply js/poloniex.js]: No worries, not having instanceof at hand is counter-intuitive indeed. I know that we can transpile this to Python, not sure about PHP yet, looking up.

Comment (mkutny) [reply js/poloniex.js]: If it won't we'll be able to go with something like `this.instanceOf (exception, class)`, what do you think?

Comment (kroitor) [reply js/poloniex.js]: @mkutny sure! I'm still determined to transpile it to natural language syntax, but we have alternative options just in case.

**Comment by kroitor** [comment]
Btw, just wanted to ask an offtopic question, if, maybe, you know of a way to make linter tolerate `"\0"` and `"\n"` (PHP needs them in double-quotes, single-quoted special chars are not expanded). If it isn't possible, I'll hide these constants in the base class to satisfy the linter.

**Comment by mkutny** [comment]
```javascript
            // unfortunately, PHP demands double quotes for the escaped newline symbol
            // eslint-disable-next-line quotes
            let payload = [ method, this.hostname, url, auth ].join ("\n");
```

**Comment by kroitor** [comment]
@mkutny awesome!

**Comment by Sukaesihessy** [comment]
Terima kasih konfirmasinya

**Pull request merged**